image-hd.c

```diff
@@ -511,6 +511,12 @@ static int hdimage_setup(struct image *image, cfg_t *cfg)
 		if (part->in_partition_table)
 			++partition_table_entries;
 	}
+	if (partition_table_entries > 0 && !hd->partition_table) {
+		image_error(image, "partition setting 'in-partition-table = true' "
+				"is not compatible with "
+				"hdimage setting 'partition-table = false'\n");
+		return -EINVAL;
+	}
```

Comment:
Hmm, I just noticed that this does not make sense right now. The problem is that `in-partition-table` defaults to `true`. So with `partition-table = false` all partitions need an explicit `in-partition-table = false`.
For this check to work properly, we would need to change the default dynamically depending on `partition-table`. I thinks that's possible, but I'm not sure if It's worth the effort.

Reply:
Hm, yeah, the default is hard to modify, and there's no way to ask libconfuse "is this the default value or something explicitly specified". The "partition-table = false" case is probably quite rare (there's no tests doing that AFAICT), so perhaps one should simply ignore "in-partition-table" (i.e. force set it to false) when partition-table is false. That will make most configuration files easier to read since one doesn't need all the explicit "in-partition-table = true/false", and the only thing we lose is a sanity check for somebody that explicitly says "in-partition-table=true" and "partition-table=false".

Reply:
Sounds good.